src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -142,18 +149,33 @@ export class CdkVirtualScrollViewport extends CdkScrollable implements OnInit, O
   /** A list of functions to run after the next change detection cycle. */
   private _runAfterChangeDetection: Function[] = [];
 
+  /** Subscription to changes in the viewport size. */
+  private _viewportChanges = Subscription.EMPTY;
+
   constructor(public elementRef: ElementRef<HTMLElement>,
               private _changeDetectorRef: ChangeDetectorRef,
               ngZone: NgZone,
               @Optional() @Inject(VIRTUAL_SCROLL_STRATEGY)
                   private _scrollStrategy: VirtualScrollStrategy,
               @Optional() dir: Directionality,
-              scrollDispatcher: ScrollDispatcher) {
+              scrollDispatcher: ScrollDispatcher,
+              /**
+               * @deprecated `viewportRuler` parameter to become required.
```

Comment:
I would just mark this `@breaking-change`, not `@deprecated` since the change is making it required rather than removing it

Reply:
I’ve been marking these as deprecated in other places, because it’s something standard that’ll show up in people’s tooling, whereas breaking-change is something we made up.

Reply:
I see, I'm not sure if this will show up the way we want though, it will probably show `viewportRuler` as a deprecated param. What might work better is TS's method overloading. We can mark the old signature deprecated and the new one not deprecated.

Obviously we already have this same pattern all over the code base, but it might be nice to start doing it the way I suggested going forward, I'll leave it up to you

Reply:
Good point that we should be using overloads instead, but I realized that Ivy won't generate the correct code if the constructor is overloaded with a different number of parameters. I've submitted a PR to fix the compiler here https://github.com/angular/angular/pull/34590 so until that gets released I'll keep it as is.